### ISSUE TITLE
`test_distributions_random` failing due to change in `st.chisquare` behavior

### ISSUE THREAD

**Issue by ricardoV94** [pymc-devs/pymc#4810]
The tests in `test_distributions_random` which rely on `pymc3_random_discrete` started failing in the `v3` branch due to changes in the behavior of `st.chisquare`. I don't think any test is making use of this helper in the `main` branch after the `v4` refactoring, so we are not seeing failures related to that here.

For instance, in the `v3` branch, `test_distributions_random.py::TestScalarParameterSamples::test_poisson` passes locally with scipy `1.6.3` but fails with `1.7.0`

It raises a `ValueError` in `st.chisquare`:

```python
            if diff_gt_tol:
                msg = (f"For each axis slice, the sum of the observed "
                       f"frequencies must agree with the sum of the "
                       f"expected frequencies to a relative tolerance "
                       f"of {rtol}, but the percent differences are:\n"
                       f"{relative_diff}")
>               raise ValueError(msg)
E               ValueError: For each axis slice, the sum of the observed frequencies must agree with the sum of the expected frequencies to a relative tolerance of 1e-08, but the percent differences are:
E               0.006036217303822937
```
This issue / breaking change was added in https://github.com/scipy/scipy/pull/13193 and mentioned in https://github.com/scipy/scipy/issues/13362 and https://github.com/scipy/scipy/issues/14298

It seems like we need to normalize the inputs as the behavior of `st.chisquare` changed.

_Originally posted by @ricardoV94 in https://github.com/pymc-devs/pymc3/pull/4808#issuecomment-869192843_

**Comment by maresb** [comment]
I have not yet dug into this issue. Does the problem affect purely the test case or also the pymc3 codebase? It seems that there is also `numpy.random.Generator.chisquare` which maybe gets called [here](https://github.com/pymc-devs/pymc3/blob/77ce2005e2fce00ba28d9fd3ccaeb79eb6fa3770/pymc3/distributions/multivariate.py#L280)? Could that also be affected?

In case the codebase is also affected, we may need to adjust the conda-forge recipe to pin `scipy <1.7.0` for `pymc3 <3.11.3`

**Comment by ricardoV94** [comment]
It is just the test case that uses `scipy.stats.chisquare` (the [statistical test](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.chisquare.html), not the [probability distribution](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.chi2.html))

**Comment by fonnesbeck** [comment]
The error is coming from `test_multinomial`, where the sums of the samples (expected and observed) are not always the same:

```
    def test_multinomial(self):
        def ref_rand(size, p, n):
            return nr.multinomial(pvals=p, n=n, size=size)
    
        for n in [2, 3]:
>           pymc3_random_discrete(
                pm.Multinomial,
                {"p": Simplex(n), "n": Nat},
                valuedomain=Vector(Nat, n),
                size=100,
                ref_rand=ref_rand,
            )

[[199 200]
 [ 99 100]]
```
Previous versions of the SciPy function did not check this.

**Comment by michaelosthege** [comment]
Closed by #4840
